gcc/rust/rust-diagnostics.h

```diff
@@ -185,9 +230,22 @@ struct Error
   }
 
   // TODO: the attribute part might be incorrect
+  // simple location
   Error (location_t locus, const char *fmt,
 	 ...) /*RUST_ATTRIBUTE_GCC_DIAG (2, 3)*/ RUST_ATTRIBUTE_GCC_DIAG (3, 4);
```

Comment:
Just eager to know, I tried using the commented version of `RUST_ATTRIBUTE_GCC_DIAG` macro (they are seem to correct). But don't know why are not working

Maybe they are linked with these constructor definitions:
https://github.com/Rust-GCC/gccrs/blob/5712d7d383b0597cd00466e5995c1e620c2fc809/gcc/rust/rust-diagnostics.cc#L387-L390

Or is there any other logic or reason?